backend/functions/src/middleware/auth-middleware.ts

```diff
@@ -43,10 +44,50 @@ export function checkAuth(req: Request, res: Response, next: NextFunction) {
   }
 }
 
+export function logReqBody(req: Request, res: Response, next: NextFunction) {
+  // TODO (richardgu): find a way to also log which endpoint is being hit
+  let output = `Request Endpoint: ${req.originalUrl}  `
+  if (Object.keys(req.params).length > 0) {
+    output += 'REQUEST PARAMS: {'
+    let ind = 0
+    for (const key in req.params) {
+      const val = req.params[key]
+      output += `${JSON.stringify(key).substring(
+        1,
+        JSON.stringify(key).length - 1
+      )}: ${JSON.stringify(val).substring(1, JSON.stringify(val).length - 1)}`
+      if (ind < Object.keys(req.params).length - 1) output += ', '
+      ind++
+    }
+    output += '}'
+  } else {
+    output += 'Request params is empty.  '
+  }
```

Comment:
I don't think you need these sets of if/elses. If the params are empty, I think you can just use a visual reqpresentation of something being empty, like `{}`.